pkg/expansion/expand.go

```diff
@@ -0,0 +1,102 @@
+package expansion
```

Comment:
package documentation.

And while we're at it, can you break this out into a separate PR that we can merge first?

It's a neat little self contained package plus test.

Reply:
I put the expansion package in PR #56